### PR TITLE
Fix addition of PauliSumOp with parameters

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

The following code
```
a = Parameter('a')
b = Parameter('b')
a * PauliSumOp.from_list([('X', 2)]) + b * PauliSumOp.from_list([('Y', 1)])
```
raises the error `QiskitError: 'other is not a number'` because multiplication between SparsePauliOp and ParameterExperession is not supported.

In this PR, I avoid this error by the workaround until https://github.com/Qiskit/qiskit-terra/issues/6086, and it returns SummedOp of PauliSumOp.

### CODE REVIEW

File: qiskit/opflow/primitive_ops/pauli_sum_op.py
Comment (Cryoris) [inline qiskit/opflow/primitive_ops/pauli_sum_op.py:119]: Does this cover all numeric types? Wouldn't it be safer to check the other way around, if it is not a `ParameterExpression`?

Comment (ikkoham) [reply qiskit/opflow/primitive_ops/pauli_sum_op.py:119]: As you can see from the type hints, we do not support other types. If we need support, should I rewrite everything else to support numpy.integer, floating, complexfloating (e.g. in OperatorBase.mul())?

Comment (Cryoris) [reply qiskit/opflow/primitive_ops/pauli_sum_op.py:119]: Technically you're correct, but I think we can be a bit more accommodating for numeric types and support all kinds of numbers. I think when we say "we support (int, float, complex)" that implicitly stands for any number. Sometimes users accidentally use a numpy type without knowing it (or I do 😄). 

Thus it's much safer to check the other way around, for a `ParameterExpression`.

Comment (jlapeyre) [reply qiskit/opflow/primitive_ops/pauli_sum_op.py:119]: If the `_multiply` method in `SparsePauliOp` will accept any `Number`, then we should allow sending any `Number`, right? 
Isn't this independent of the specification of `OperatorBase.mul`?

I support the liberal route, checking for `ParameterExpression`. (I did this in the fix I provided for the user with the problem: https://github.com/jlapeyre/qiskit-terra/tree/pauli-sum-op-add)

Or maybe better, checking if we are sending a `Number`, since that is what `SparsePauliOp._multiply` claims to support.

Comment (ikkoham) [reply qiskit/opflow/primitive_ops/pauli_sum_op.py:119]: OK, we don't officially support numpy's numeric types for now. I also prefer much safer way. So I will follow your suggestion.

@jlapeyre Thanks. Sorry, my wording was bad. `PrimitiveOps.coeff` is assumed to be `int, float, complex or ParameterExpression`. So, passing numpy's numeric types becomes undefined behavior (it may be work).

**Review by woodsp-ibm — DISMISSED** [review]

File: releasenotes/notes/fix-addition-parameterised-paulisumop-e9b859bca1e58ddd.yaml
Comment (woodsp-ibm) [inline releasenotes/notes/fix-addition-parameterised-paulisumop-e9b859bca1e58ddd.yaml:6]:
```suggestion
    Addition between :class:`~qiskit.opflow.PauliSumOp`s that had
    :class:`~qiskit.circuit.ParameterExpression` coefficients raised `QiskitError`.
    Now we support the addition.
```

Comment (woodsp-ibm) [reply releasenotes/notes/fix-addition-parameterised-paulisumop-e9b859bca1e58ddd.yaml:6]: I changed it slightly as 'raises' still felt like the present tense where it still might do it.

**Comment by ikkoham** [comment]
Mergeable?

**Review by woodsp-ibm — APPROVED** [review]

**Comment by ikkoham** [comment]
@Qiskit/terra-core Could you merge this PR if possible? I can't understand why the automerge doesn't work.

**Comment by woodsp-ibm** [comment]
It needs a second approval, i.e from terra-core team, once that is done then it should be fine (maybe I should not have added automerge as soon as I did)

**Comment by ikkoham** [comment]
@woodsp-ibm Thanks. I thought one code owner approval is enough.

**Review by Cryoris — APPROVED** [review]

**Pull request merged**
